gateway/bin/apicast

```diff
@@ -109,3 +109,61 @@ for my $inc ($apicast_src, 'src') {
 my @args = ('resty', @resty_args, $bin, @ARGV);
 
 exec '/usr/bin/env', @args;
+
+__DATA__
+#!/usr/bin/env resty
```

Comment:
I think we need an explanation here about why this is needed. Like the one included in the PR description.

Reply:
Added more comments. Most of them are in the lua helper code.